### PR TITLE
ci: Preliminary support for GitHub merge queues

### PR DESCRIPTION
**Pull request by blampe**
It occurred to me that @AaronFriel had setup this repo to test Bors. It would make sense to also use it as a guinea pig for merge queues before landing https://github.com/pulumi/pulumi/pull/13681.

### CODE REVIEW

**Comment by abhinav** [comment]
RE: lint failure: actionlint probably needs to be updated https://github.com/pulumi/watchutil-rs/blob/main/.github/workflows/ci-lint.yml#L25C18-L25C45

**Review by abhinav — APPROVED** [review]

**Comment by blampe** [comment]
Current dependencies on/for this PR:
* main
  * **PR #37** <a href="https://app.graphite.dev/github/pr/pulumi/watchutil-rs/37" target="_blank">Graphite</a>  👈
    * **PR #38** <a href="https://app.graphite.dev/github/pr/pulumi/watchutil-rs/38" target="_blank">Graphite</a> 
      * **PR #39** <a href="https://app.graphite.dev/github/pr/pulumi/watchutil-rs/39" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/pulumi/watchutil-rs/37?utm_source=stack-comment).

**Comment by blampe** [comment]
bors merge

**Comment by blampe** [comment]
bors retry

**Comment by blampe** [comment]
bors retry

**Comment by blampe** [comment]
bors merge

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,11 +31,11 @@ jobs:
 
           ./.github/scripts/set-output version "${VERSION}"
       - name: Install git-cliff
-        uses: baptiste0928/cargo-install@bf6758885262d0e6f61089a9d8c8790d3ac3368f # v1.3.0
```
Comment (AaronFriel) [inline .github/workflows/ci.yml:33]: Curious why this is unpinned? Best practice is to [pin to a commit SHA](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions).

Comment (blampe) [reply .github/workflows/ci.yml:33]: @AaronFriel I'm just trying to get the build passing, seems it's been red for a while. I'll change it.

Comment (blampe) [reply .github/workflows/ci.yml:33]: @AaronFriel the remaining [failure](https://github.com/pulumi/watchutil-rs/actions/runs/5814345665/job/15766330788) is on the release step, and I'm pretty out of my element. It seems like pu/pu migrated to cosign in https://github.com/pulumi/pulumi/pull/11310, so my plan was to try porting that over to this repo. Would love to hear if you have alternatives before I go down that path.

Comment (blampe) [reply .github/workflows/ci.yml:33]: Discussed 1:1, I've disabled the signing step.

**Comment by blampe** [comment]
bors merge

**Pull request merged**